### PR TITLE
Summary components

### PR DESCRIPTION
**Pull request by Mussuranae**
New shooting information summary component
New sales pitch summary component

### CODE REVIEW

File: apps/financiers/financiers/src/app/dashboard/tunnel/summary/summary.component.html [hunk 1]
```diff
@@ -73,6 +73,24 @@ <h3>Additional Information</h3>
 
     <mat-divider></mat-divider>
 
+    <!-- SHOOTING INFORMATION -->
+
+    <section id="shooting-information">
```
Comment (GrandSchtroumpf) [inline apps/financiers/financiers/src/app/dashboard/tunnel/summary/summary.component.html:78]: Verify if it's the right status. There is a pipe `hasStatus` for that.

Comment (GrandSchtroumpf) [reply apps/financiers/financiers/src/app/dashboard/tunnel/summary/summary.component.html:78]: Same for all the other sections

File: libs/movie/src/lib/movie/form/movie.form.ts [hunk 2]
```diff
@@ -529,18 +528,17 @@ export class FilmographyForm extends FormEntity<FilmographyFormControl> {
 }
 
 // ------------------------------
-//          TOTALBUDGET
+//          TOTAL BUDGET
 // ------------------------------
 
 function createTotalBudgetFormControl(totalBudget?: Partial<MovieTotalBudget>) {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/movie.form.ts:534]: You need to set a default value : 
`createTotalBudgetFormControl(totalBudget: Partial<MovieTotalBudget> = {})`

File: libs/movie/src/lib/movie/form/summary/shooting-information/shooting-information.component.html [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    <li *ngIf="!shootingDates.get('completed').value && !shootingDates.get('progress').value && !isThereShootingPlanned()">
+      <strong>Shooting Dates </strong>
+      <a [routerLink]="link" fragment="shooting-dates" class="missing-control">
+        <span>Missing</span>
+      </a>
+    </li>
+    <li *ngIf="movie.shooting.get('dates').get('completed').value">
+      <strong>Shooting Completed </strong>
+      <missing-control [control]="movie.shooting.get('dates').get('completed')" [link]="link" fragment="shooting-dates">
+        <ng-template #content>
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/summary/shooting-information/shooting-information.component.html:17]: Why do you need a #content ?

File: libs/movie/src/lib/movie/form/summary/shooting-information/shooting-information.component.html [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+          {{ movie.shooting.get('dates').get('completed').value | date: 'shortDate' }}
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/summary/shooting-information/shooting-information.component.html:18]: I think you can do : 
```html
<ng-template let-completed>
  {{ completed| date: 'shortDate' }}
</ng-template>
```
If so, do it everywhere else.

File: libs/movie/src/lib/movie/form/summary/shooting-information/shooting-information.component.html [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        <missing-control [control]="shootingPlanned.get('to').get('year')" [link]="link" fragment="shooting-dates">
+          <ng-template #content>
+            <span>{{ shootingPlanned.get('to').get('year').value }}</span>
+          </ng-template>
+        </missing-control>
+      </span>
+    </li>
+    <li>
+      <strong>Shooting Locations </strong>
+      <ng-container *ngIf="!shootingLocations.controls.length else showShootingLocations">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/summary/shooting-information/shooting-information.component.html:68]: use the missing-controls here. Max updated it to support FormArray & FormGroup.
Do the same below

File: libs/movie/src/lib/movie/form/summary/shooting-information/shooting-information.component.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import { Component, Input } from '@angular/core';
+import { MovieForm } from '@blockframes/movie/form/movie.form';
+
+@Component({
+  selector: '[movie][link] movie-summary-shooting-information',
+  templateUrl: './shooting-information.component.html',
+  styleUrls: ['./shooting-information.component.scss']
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/summary/shooting-information/shooting-information.component.ts:7]: changedetectionstrategy.onpush

File: libs/movie/src/lib/movie/form/summary/shooting-information/shooting-information.component.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+  get shootingLocations() {
+    return this.movie.shooting.get('locations');
+  }
+
+  isThereShootingPlanned() {
+    const shootingFrom = this.shootingPlanned.get('from');
+    const shootingTo = this.shootingPlanned.get('to');
+
+    const noFrom = !!shootingFrom.get('period').value || !!shootingFrom.get('month').value || !!shootingFrom.get('year').value;
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/summary/shooting-information/shooting-information.component.ts:30]: This is super difficult to read: 
```typescript
isThereShootingPlanned() {
  const exists = (key) => {
    const { period, month, year } = this.shootingPlanned.get(key).value;
    return period || month || year;
  }
  return exists('from') || exists('to');
}
```

**Review by GrandSchtroumpf — CHANGES_REQUESTED** [review]

File: libs/movie/src/lib/movie/form/summary/shooting-information/shooting-information.component.ts (hunk 7 above)
Comment (Mussuranae) [reply libs/movie/src/lib/movie/form/summary/shooting-information/shooting-information.component.ts:30]: Oh thank you, it was a mess to write this xD

File: libs/movie/src/lib/movie/form/summary/sales-pitch/sales-pitch.component.html [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    <li>
+      <missing-control [control]="movie.goals" [link]="link" fragment="audience-goals">
+        <ng-template let-audienceGoals>
+          <strong>Target Audiance </strong>
+          <ng-container *ngFor="let goal of audienceGoals; let last = last">
+            <span>{{ goal.target | titlecase }}{{ last ? '' : ', ' }}</span>
+          </ng-container>
+          <br/>
+          <strong>Social Responsability Goals </strong>
+          <ng-container *ngFor="let goal of audienceGoals; let last = last">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/summary/sales-pitch/sales-pitch.component.html:35]: Normally toLabel should support arrays like translateSlug. If not let me know: 
`{{ audienceGoals | deepKey:'goal' | toLabel: 'socialGoals' }}`

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**